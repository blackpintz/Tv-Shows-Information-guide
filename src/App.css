.img {
  width: 100%;
}

.detail-img {
  width: 470px;
  height: 555px;
}

.summary {
  width: 800px;
}
