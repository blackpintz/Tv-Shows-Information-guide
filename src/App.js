import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import propTypes from 'prop-types';
import './App.css';
import TvShowList from './components/TvShowList';
import TvShowDetails from './components/TvShowDetails';

const App = ({ tvShows }) => {
  console.log(tvShows);
  return (
    <div className="App">
      <Switch>
        <Route exact path="/" component={TvShowList} />
        <Route exact path="/show/:id" render={routeProps => <TvShowDetails id={routeProps.match.params.id} routeProps={routeProps} />} />
      </Switch>
    </div>
  );
};

const mapStateToProps = ({ tvShows }) => ({
  tvShows,
});

App.propTypes = {
  tvShows: propTypes.arrayOf(propTypes.object),
};

App.defaultProps = {
  tvShows: 'no value',
};

export default connect(mapStateToProps)(App);
