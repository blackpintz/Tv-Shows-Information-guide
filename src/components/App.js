import React from 'react';
import { Route, Switch } from 'react-router-dom';
import 'bootstrap/dist/css/bootstrap.min.css';
import '../App.css';
import connectTvShowList from '../containers/TvShowList';
import ConnectTvShowDetails from '../containers/TvShowDetails';

const App = () => (
  <div className="App">
    <Switch>
      <Route exact path="/" component={connectTvShowList} />
      <Route exact path="/show/:id" render={routeProps => <ConnectTvShowDetails id={routeProps.match.params.id} routeProps={routeProps} />} />
    </Switch>
  </div>
);

export default App;
