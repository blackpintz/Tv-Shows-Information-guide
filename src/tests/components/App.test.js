import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import App from '../../App';

test('renders App correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<App />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
