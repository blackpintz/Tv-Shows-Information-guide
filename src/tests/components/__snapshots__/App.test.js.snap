// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly 1`] = `
<div
  className="App"
>
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Route
      exact={true}
      path="/show/:id"
      render={[Function]}
    />
  </Switch>
</div>
`;
